src/de/dkfz/b080/co/common/BasicCOProjectsRuntimeService.groovy

```diff
@@ -251,11 +299,11 @@ class BasicCOProjectsRuntimeService extends RuntimeService {
         return new File(getInpDirectory(COConstants.CVALUE_SEQUENCE_DIRECTORY, process, sample, library).getAbsolutePath().replace('${run}', run));
     }
 
-    BasicBamFile getMergedBamFileForDataSetAndSample(ExecutionContext context, Sample sample) {
-        return getMergedBamFileForDataSetAndSample(context, null, sample);
+    BasicBamFile getMergedBamFileFromFilesystem(ExecutionContext context, Sample sample) {
```

Comment:
Remove this method and set dataSet to null by default in the next method?

Reply:
O.k. Note, however, I still dislike null -- also as parameter. The only way to figure out, whether a method accepts null for a parameter value is trial and error or reading the code. Both significantly hamper development speed. Better is to have the interface express this. Either by not having the parameter, or by using Optional.